apps/src/blockTooltips/DropletAutocompleteParameterTooltipManager.js

```diff
@@ -88,15 +88,21 @@ DropletAutocompleteParameterTooltipManager.prototype.onCursorMovement_ = functio
 DropletAutocompleteParameterTooltipManager.prototype.showParamDropdownIfNeeded_ = function (editor, paramInfo) {
   // Check the dropletConfig to see if we can find dropdown info for this parameter
   var dropdownList;
+  // TODO: Use getAllAvailableDropletBlocks() if we need dropdowns on
+  // builtinConfigBlocks or standardConfig blocks
   this.dropletTooltipManager.dropletConfig.blocks.forEach(function (block) {
-    if (block.func === paramInfo.funcName && block.dropdown) {
+    if (!block.noAutocomplete &&
+        (!this.dropletTooltipManager.blockFilter ||
+          typeof this.dropletTooltipManager.blockFilter[block.func] !== 'undefined') &&
+        block.func === paramInfo.funcName &&
+        block.dropdown) {
```

Comment:
Between the number of different conditionals we have, and the number of negatives, this is getting to be pretty difficult to parse. Can we simplify?

Seems like one of the first things we could do is
```
if (!block.dropdown || block.noAutocomplete) {
  return;
}
```
It's not clear to me what `block.func === paramInfo.funcName` means exactly.

`typeof this.dropletTooltipManager.blockFilter[block.func] !== 'undefined'` can just be `this.dropletTooltipManager.blockFilter[block.func] !== undefined`
(the only time that doesn't work is if we're checking for the existence of a local variable instead of a member variable, i.e. if i do `flkjsdfa === undefined` i can get an exception, but if I do `foo.flkjsdfa === undefined` i cant)